scripts/build-semgrep-proprietary

```diff
@@ -0,0 +1,46 @@
+#! /usr/bin/env bash
+#
+# CI job that checks whether semgrep-proprietary can compile using the
+# current semgrep commit as a submodule.
+#
+set -eu
+
+semgrep_pro_url="git@github.com:returntocorp/semgrep-proprietary.git"
+semgrep_root=$(git rev-parse --show-toplevel)
+semgrep_commit=$(git rev-parse --short=7 HEAD)
+
+echo <<EOF
+We're attempting to build semgrep-proprietary using the current version
+of semgrep. We're assuming we're in a folder within the semgrep repo.
+
+Here's what we found about semgrep:
+
+semgrep repo root: $semgrep_root
+semgrep commit: $semgrep_commit
+semgrep-proprietary remote URL: $semgrep_pro_url
+semgrep git status:
+$(cd "$semgrep_root" && git status)
+EOF
+
+# Use a folder where it's safe to delete any existing semgrep-proprietary.
+workdir=$semgrep_root/../tmp-semgrep-pro
+mkdir -p "$workdir"
+(
+  cd "$workdir"
+  rm -rf semgrep-proprietary
+  git clone "$semgrep_pro_url" --depth 1
```

Comment:
I think auth might need to be configured here, since `semgrep-proprietary` is a private repo.